pkg/reconciler/revision/resources/queue.go

```diff
@@ -270,3 +313,37 @@ func makeQueueContainer(rev *v1alpha1.Revision, loggingConfig *logging.Config, o
 		}},
 	}
 }
+
+func applyReadinessProbeDefaults(p *corev1.Probe, port int32) {
+	switch {
+	case p == nil:
+		return
+	case p.HTTPGet != nil:
+		p.HTTPGet.Host = localAddress
+		p.HTTPGet.Port = intstr.FromInt(int(port))
+
+		if p.HTTPGet.Scheme == "" {
+			p.HTTPGet.Scheme = corev1.URISchemeHTTP
+		}
+
+		p.HTTPGet.HTTPHeaders = append(p.HTTPGet.HTTPHeaders, corev1.HTTPHeader{
+			Name:  network.KubeletProbeHeaderName,
+			Value: queue.Name,
+		})
+	case p.TCPSocket != nil:
+		p.TCPSocket.Host = localAddress
+		p.TCPSocket.Port = intstr.FromInt(int(port))
+	case p.Exec != nil:
+		// Use default TCP connect probe to ensure data path is open from queue-proxy to
```

Comment:
I see why you want to keep the TCP probe in this case, but I feel like it's not really helping us in ensuring the user-container actually does the right thing. After all, we don't check if it accepts HTTP/HTTP2. Should we remove this defaulting for simplicity?

Reply:
@joshrider This is sort of a gray area, so let's remove this and open an issue to discuss and follow up?